pkg/project/registry/project/proxy/proxy.go

```diff
@@ -154,19 +158,38 @@ func (s *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectI
 		return nil, false, fmt.Errorf("not a project: %#v", obj)
 	}
 
-	s.updateStrategy.PrepareForUpdate(ctx, obj, oldObj)
-	if errs := s.updateStrategy.ValidateUpdate(ctx, obj, oldObj); len(errs) > 0 {
-		return nil, false, kerrors.NewInvalid(projectapi.Kind("Project"), project.Name, errs)
+	user, ok := kapi.UserFrom(ctx)
+	if !ok {
+		return nil, false, kerrors.NewForbidden(projectapi.Resource("project"), project.Name, fmt.Errorf("unable to update project without a user on the context"))
 	}
 
-	namespace, err := s.client.Update(projectutil.ConvertProject(project))
+	var namespace *kapi.Namespace
+	namespace, err = s.getImpersonatingClientFor(user).Update(projectutil.ConvertProject(project)) // Try to update namespace directly while impersonating the user
 	if err != nil {
-		return nil, false, err
+		// If updating the namespace as the user fails, validate their changes and then try to update namespace as the privileged client
+		s.updateStrategy.PrepareForUpdate(ctx, obj, oldObj)
+		if errs := s.updateStrategy.ValidateUpdate(ctx, obj, oldObj); len(errs) > 0 {
+			return nil, false, kerrors.NewInvalid(projectapi.Kind("Project"), project.Name, errs)
+		}
+
+		namespace, err = s.client.Update(projectutil.ConvertProject(project))
+		if err != nil {
+			return nil, false, err
+		}
 	}
 
 	return projectutil.ConvertNamespace(namespace), false, nil
 }
 
+func (s *REST) getImpersonatingClientFor(user user.Info) kclient.NamespaceInterface {
```

Comment:
that reconstructs from config on every request, potentially down to the point of re-reading and re-parsing CA and cert files... needs to be split up to only build the base transport once